app/controllers/results_controller.rb

```diff
@@ -320,6 +320,16 @@ def next_grouping
     render json: { next_result: next_result, next_grouping: next_grouping }
   end
 
+  def next_random_unmarked_grouping
+    result = record
+    grouping = result.submission.grouping
+
+    next_grouping = grouping.get_next_random_unmarked(current_role)
+    next_result = next_grouping&.current_result
+
+    render json: { next_result: next_result, next_grouping: next_grouping }
```

Comment:
Don't send the full object JSON, just the specific pieces of data that are necessary (result/submission/grouping id).